target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -324,6 +324,12 @@ fn deserialize_guardian_set_checked(
         ReceiverError::InvalidGuardianSetPda
     );
 
+    let timestamp = Clock::get().map(Into::into)?;
+    require!(
+        guardian_set.inner().is_active(&timestamp),
+        ReceiverError::GuardianSetExpired
+    );
+
```

Comment:
I forgot this check 🤦

Reply:
On-chain benchmarks users don't use a data way in the past. if we want to be accurate we can pass the price timestamp here but i don't think it's necessary.

Reply:
i think this is ok for now. we haven't solved this problem on any of the other chains either.